### PR TITLE
Add AWS Secret manager 

### PR DESCRIPTION
**Pull request by josenava**
With this PR we will be able to get the env vars from AWSSecret manager

### CODE REVIEW

File: ecs_deployer/conf.py [hunk 1]
```diff
@@ -5,6 +5,9 @@
 VAULT_TOKEN = os.getenv('VAULT_TOKEN', None)
 VAULT_PATH = os.getenv('VAULT_PATH', None)
 
+AWS_SECRET_NAME = os.getenv('AWS_SECRET_NAME', None)
+AWS_REGION_NAME = os.getenv('AWS_REGION_NAME', None)
```
Comment (jairovadillo) [inline ecs_deployer/conf.py:9]: The default value for `getenv` is None so there's no need for this second arg

**Review by jairovadillo — APPROVED** [review]
🔝  usage of ABC

File: ecs_deployer/secret_managers.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        return [{"name": k, "value": v} for k, v in env_vars.items()]
+
+
+class AWSSecretManager(SecretsManager):
+    def __init__(self, secret_name: str, region_name: str):
+        self.secret_name = secret_name
+        self.region_name = region_name
+
+    @classmethod
+    def build(cls, manager_config: Dict[str, Any]):
```
Comment (guillem21b) [inline ecs_deployer/secret_managers.py:53]:
```suggestion
    def build(cls, manager_config: Dict[str, Any]) -> 'AWSSecretManager':
```

File: ecs_deployer/secret_managers.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+class VaultManager(SecretsManager):
+
+    def __init__(self, host: str, token: str, path: str):
+        self.host = host
+        self.token = token
+        self.path = path
+
+    @classmethod
+    def build(cls, vault_manager_config: Dict[str, Any]):
```
Comment (guillem21b) [inline ecs_deployer/secret_managers.py:26]:
```suggestion
    def build(cls, vault_manager_config: Dict[str, Any]) -> 'VaultManager':
```

File: ecs_deployer/factories.py [hunk 4]
```diff
@@ -1,7 +1,11 @@
-from vault import VaultManager
+from typing import Any, Dict
+from secret_managers import AWSSecretManager, VaultManager
 
 
-def build_secrets_manager(driver, secrets_manager_config):
+def build_secrets_manager(driver: str, secrets_manager_config: Dict[str, Any]):
```
Comment (guillem21b) [inline ecs_deployer/factories.py:5]:
```suggestion
def build_secrets_manager(driver: str, secrets_manager_config: Dict[str, Any]) -> Union[AWSSecretManager, VaultManager]:
```

File: ecs_deployer/factories.py [hunk 5]
```diff
@@ -1,7 +1,11 @@
-from vault import VaultManager
+from typing import Any, Dict
```
Comment (guillem21b) [inline ecs_deployer/factories.py:1]:
```suggestion
from typing import Any, Dict, Union
```

**Review by guillem21b — COMMENTED** [review]
moar typing

File: ecs_deployer/factories.py (hunk 4 above)
Comment (josenava) [reply ecs_deployer/factories.py:5]: IMHO, this one is a bit different, it should be `-> Type[SecretManager]` as they both inherit from SecretManager.

I was going to make another PR fixing all of those PR and adding more to the methods missing documentation, so I will skip this one and push the others you suggested

File: ecs_deployer/conf.py (hunk 1 above)
Comment (josenava) [reply ecs_deployer/conf.py:9]: These lines will be fixed in another PR where we will _beautify_ the code removing these unnecessary Nones and add more typing to document better the methods

**Review by GuillemSerra — APPROVED** [review]

**Pull request merged**